src/main/java/hudson/remoting/Channel.java

```diff
@@ -327,6 +327,12 @@
     /*package*/ @Nonnull final ClassFilter classFilter;
 
     /**
+     * Indicates that close of the channel has been requested.
+     * After it it does not make sense to execute new user-space commands.
+     */
+    private boolean closeRequested = false;
```

Comment:
Could we drop this field and just check `outClosed != null`, since `close` sets both? Or is there a race condition in doing that?